### PR TITLE
optimize(hz): handler_single.go supports more template parameters

### LINKED ISSUE #1001: [QUESTION] hz handler.go 与 handler_single.go传值不一样
handler.go
https://github.com/cloudwego/hertz/blob/c849a7a38e42c4159d8ab12a1598dfbe5e080b3f/cmd/hz/generator/handler.go#L185C1-L187C3

handler_single.go
https://github.com/cloudwego/hertz/blob/c849a7a38e42c4159d8ab12a1598dfbe5e080b3f/cmd/hz/generator/handler.go#L237C1-L245C52

2个地方传的参数不一致，导致我想_single.go使用PackageName失败。
```
layouts:
  - path: handler_single.go
    delims:
      - '{{'
      - '}}'
    body: |+
      {{.Comment}}
      func {{$.Name}}(ctx context.Context, c *app.RequestContext) {
      	var err error
      	{{if ne .RequestTypeName "" -}}
      	var req {{$.RequestTypeName}}
      	err = c.BindAndValidate(&req)
      	if err != nil {
      		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
      		return
      	}
      	{{end}}
      	resp := {{$.PackageName}}service.New{{$.PackageName | camelcase}}Service(ctx, c).{{$.Name}}(&req)

      	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
      }
```

是否可以考虑优化成如下：
```
        # 拷贝hander里面值
	baseData := make(map[string]string)
	keys := reflect.TypeOf(handler)
	values := reflect.ValueOf(handler)
	for i := 0; i < keys.NumField(); i++ {
		if values.Field(i).Kind() == reflect.String {
			baseData[keys.Field(i).Name] = values.Field(i).String()
		}
	}
	// insert new handler
	for _, method := range handler.(Handler).Methods {
		if bytes.Contains(file, []byte(fmt.Sprintf("func %s(", method.Name))) {
			continue
		}

		// Generate additional handlers using templates
		handlerSingleTpl := pkgGen.tpls[handlerSingleTplName]
		if handlerSingleTpl == nil {
			return fmt.Errorf("tpl %s not found", handlerSingleTplName)
		}
		data := make(map[string]string)
		for k, v := range baseData {
			data[k] = v
		}
		data["Comment"] = method.Comment
		data["Name"] = method.Name
		data["RequestTypeName"] = method.RequestTypeName
		data["ReturnTypeName"] = method.ReturnTypeName
		data["Serializer"] = method.Serializer
		data["OutputDir"] = method.OutputDir
		handlerFunc := bytes.NewBuffer(nil)
		err = handlerSingleTpl.Execute(handlerFunc, data)
```

如果可行的话，我提个PR

### PR DESCRIPTION
**Pull request by ypo2478**
#### What type of PR is this?
optimize

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: handler_single.go supports more template parameters
zh(optional):

#### (Optional) Which issue(s) this PR fixes:
https://github.com/cloudwego/hertz/issues/1001

### CODE REVIEW

File: cmd/hz/generator/handler.go [hunk 1]
```diff
@@ -234,7 +242,10 @@ func (pkgGen *HttpPackageGenerator) updateHandler(handler interface{}, handlerTp
 		if handlerSingleTpl == nil {
 			return fmt.Errorf("tpl %s not found", handlerSingleTplName)
 		}
-		data := make(map[string]string, 5)
+		data := make(map[string]string)
```
Comment (FGYFFFF) [inline cmd/hz/generator/handler.go:245]: 这块的传参改一下吧。把 data 封装成下面的 struct，然后再补上你想要的 package 等信息。 这样可以和之前一致，看起来也更优雅些

```
type singleHandler struct {
      HttpMethod
      packageName string
      projPackage string 
      .......
}

```

Comment (ypo2478) [reply cmd/hz/generator/handler.go]: 已经修改好

File: cmd/hz/generator/handler.go [hunk 2]
```diff
@@ -234,13 +241,16 @@ func (pkgGen *HttpPackageGenerator) updateHandler(handler interface{}, handlerTp
...
 			return fmt.Errorf("tpl %s not found", handlerSingleTplName)
 		}
-		data := make(map[string]string, 5)
-		data["Comment"] = method.Comment
-		data["Name"] = method.Name
-		data["RequestTypeName"] = method.RequestTypeName
-		data["ReturnTypeName"] = method.ReturnTypeName
-		data["Serializer"] = method.Serializer
-		data["OutputDir"] = method.OutputDir
+		data := new(SingleHandler)
```
[the remark this answers is not in the document]
Comment (ypo2478) [reply cmd/hz/generator/handler.go]: done

**Review by FGYFFFF — APPROVED** [review]
lgtm~

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**